### DISCUSSION TITLE
Why does line serialization in Fabric export different vertex coordinates?

### DISCUSSION THREAD

**Discussion by roll20plusplus** [fabricjs/fabric.js#7921] [Q&A] [answered]
Example jsfiddle: https://jsfiddle.net/ryanmalley101/az7c8uxb/7/

var rulerLine = new fabric.Line([100, 200, 300, 400]);

console.log(rulerLine);
console.log(JSON.stringify(rulerLine));
When I just print the rulerLine object I made, the x1, x2, y1, and y2 match what was passed to the Line constructor.

When I print out the JSON serialization, it has 0s for all of those values (the formatting on the jsfiddle is bad but the coordinates are at the end of the console printout).

Why is there this discrepancy? If there's a better way to serialize the Line object so it maintains the properties of the initial console log? I'm even open to using different object types than line that could give me the same effect, other object types seem to serialize better. Happy to provide extra context.

**Comment by asturur** [comment, marked as answer]
Just because the code was written in that way. It wasn't well thought or it wasn't considered an issue.
It could be definitely improved.

**Reply by roll20plusplus** [reply]
So I should file this as an issue? Seems like it'd be an easy fix, might just branch and make a PR after its updated.

**Reply by asturur** [reply]
you can try to make a pr, you are welcome.
Try to keep it small and verify with unit tests that everything works
